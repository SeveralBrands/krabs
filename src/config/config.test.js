import mockFs from 'mock-fs';
import * as conf from './';

const configExampleAsObj = {
  tenants: [
    {
      name: 'website-1',
      domains: [
        {
          dev: 'local.website-1.com',
          stage: 'stage.website-1.com',
          prod: 'website-1.com',
        },
      ],
    },
    {
      name: 'website-2',
      domains: [
        {
          dev: 'local.website-2.com',
          stage: 'stage.website-2.com',
          prod: 'website-2.com',
        },
      ],
    },
    {
      name: 'website-3',
      domains: [
        {
          dev: 'local.website-3.com',
          stage: 'stage.website-3.com',
          prod: 'website-3.com',
        },
      ],
    },
  ],
};

const configExampleAsFn = () => ({
  tenants: [
    {
      name: 'website-1',
      domains: [
        {
          dev: 'local.website-1.com',
          stage: 'stage.website-1.com',
          prod: 'website-1.com',
        },
      ],
    },
    {
      name: 'website-2',
      domains: [
        {
          dev: 'local.website-2.com',
          stage: 'stage.website-2.com',
          prod: 'website-2.com',
        },
      ],
    },
    {
      name: 'website-3',
      domains: [
        {
          dev: 'local.website-3.com',
          stage: 'stage.website-3.com',
          prod: 'website-3.com',
        },
      ],
    },
  ],
});

const configExampleAsAsyncFn = async () => ({
  tenants: [
    {
      name: 'website-1',
      domains: [
        {
          dev: 'local.website-1.com',
          stage: 'stage.website-1.com',
          prod: 'website-1.com',
        },
      ],
    },
    {
      name: 'website-2',
      domains: [
        {
          dev: 'local.website-2.com',
          stage: 'stage.website-2.com',
          prod: 'website-2.com',
        },
      ],
    },
    {
      name: 'website-3',
      domains: [
        {
          dev: 'local.website-3.com',
          stage: 'stage.website-3.com',
          prod: 'website-3.com',
        },
      ],
    },
  ],
});

test('getTenantConfig', async () => {
  expect(await conf.getTenantConfig(configExampleAsObj)).toMatchSnapshot();
  expect(await conf.getTenantConfig(configExampleAsFn)).toMatchSnapshot();
  expect(await conf.getTenantConfig(configExampleAsAsyncFn)).toMatchSnapshot();

  try {
    await conf.getTenantConfig('Wrong config');
  } catch (e) {
    // eslint-disable-next-line
    expect(e).toMatchInlineSnapshot(
      `[Error: Unknown configuration type. Expected one of: function, object, JSON, got: string]`,
    );
  }
});

test('getCwdConfig throwing an error', () => {
  mockFs({
    '.k.config.js': {}, // wrong file name, should not be resolved
  });
  expect(conf.getCwdConfig).toThrowErrorMatchingInlineSnapshot(`"No configuration file found"`);
  mockFs.restore();
});

test('getCwdConfig on .krabs.config.js file', () => {
  mockFs({
    '.krabs.config.js': configExampleAsFn(),
  });
  expect(conf.getCwdConfig).toMatchSnapshot();
  mockFs.restore();
});

test('getCwdConfig on .krabs.js file', () => {
  mockFs({
    '.krabs.js': configExampleAsFn(),
  });
  expect(conf.getCwdConfig).toMatchSnapshot();
  mockFs.restore();
});
