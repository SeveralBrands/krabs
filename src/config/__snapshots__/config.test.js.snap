// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCwdConfig on .krabs.config.js file 1`] = `[Function]`;

exports[`getCwdConfig on .krabs.js file 1`] = `[Function]`;

exports[`getTenantConfig 1`] = `
Object {
  "tenants": Array [
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-1.com",
          "prod": "website-1.com",
          "stage": "stage.website-1.com",
        },
      ],
      "name": "website-1",
    },
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-2.com",
          "prod": "website-2.com",
          "stage": "stage.website-2.com",
        },
      ],
      "name": "website-2",
    },
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-3.com",
          "prod": "website-3.com",
          "stage": "stage.website-3.com",
        },
      ],
      "name": "website-3",
    },
  ],
}
`;

exports[`getTenantConfig 2`] = `
Object {
  "tenants": Array [
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-1.com",
          "prod": "website-1.com",
          "stage": "stage.website-1.com",
        },
      ],
      "name": "website-1",
    },
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-2.com",
          "prod": "website-2.com",
          "stage": "stage.website-2.com",
        },
      ],
      "name": "website-2",
    },
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-3.com",
          "prod": "website-3.com",
          "stage": "stage.website-3.com",
        },
      ],
      "name": "website-3",
    },
  ],
}
`;

exports[`getTenantConfig 3`] = `
Object {
  "tenants": Array [
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-1.com",
          "prod": "website-1.com",
          "stage": "stage.website-1.com",
        },
      ],
      "name": "website-1",
    },
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-2.com",
          "prod": "website-2.com",
          "stage": "stage.website-2.com",
        },
      ],
      "name": "website-2",
    },
    Object {
      "domains": Array [
        Object {
          "dev": "local.website-3.com",
          "prod": "website-3.com",
          "stage": "stage.website-3.com",
        },
      ],
      "name": "website-3",
    },
  ],
}
`;
